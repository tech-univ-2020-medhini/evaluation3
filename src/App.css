@import url('https://fonts.googleapis.com/css?family=Poppins');
.App {
  display: flex;
  flex-direction: column;
  text-align: center;
  font-family: 'Poppins';
}
.header{
  height: 4rem;
  display:flex;
  background: rgb(71,151,63);
  color: white;
  align-items: center;
  justify-content: flex-start;
  width: 100%;
}