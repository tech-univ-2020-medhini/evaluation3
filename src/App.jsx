import React, { useState } from 'react';
import './App.css';
import HomePage from './containers/HomePage';
import GenrePage from './containers/GenrePage';
import SongPage from './containers/SongPage';

function App() {
  const [isSyncClicked, setIsSyncClicked] = useState(false);
  const [isGenreClicked, setIsGenreClicked] = useState(false);
  const [genre, setGenre] = useState('');
  const [songsList, setSongsList] = useState([]);
  const [songsOfGenre, setSongsOfGenre] = useState([]);
  return (
    <div className="App">
      <header className="header" />
      <div className="body">
        {
          // eslint-disable-next-line no-nested-ternary
          isSyncClicked ? isGenreClicked ? (<SongPage songsOfGenre={songsOfGenre} genreName={genre} />)
            : (<GenrePage setIsGenreClicked={setIsGenreClicked} songsList={songsList} setSongsOfGenre={setSongsOfGenre} setGenre={setGenre} />)
            : (<HomePage setIsSyncClicked={setIsSyncClicked} setSongsList={setSongsList} />)
        }

      </div>

    </div>
  );
}

export default App;
