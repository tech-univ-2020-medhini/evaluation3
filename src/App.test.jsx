import React from 'react';
import { render } from '@testing-library/react';
import App from './App.jsx';


describe('the app page', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
