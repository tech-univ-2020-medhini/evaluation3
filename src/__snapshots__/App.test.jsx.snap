// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the app page should render correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="header"
    >
      My 
      <b>
          Record 
      </b>
      Shelf
    </div>
    <div
      class="body"
    >
      <div
        class="HomePage"
      >
        <p>
          :((
        </p>
        <p>
          Seems a bit empty in here
        </p>
        <button
          class="Button"
          data-testid="test-btn"
          type="button"
        >
          sync
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
